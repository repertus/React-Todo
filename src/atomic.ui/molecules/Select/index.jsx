import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { SelectStyling, DivStyling } from './index.styling';
import Label from '../../atoms/Label/index';

class Select extends PureComponent {
  render() {
    const { options } = this.props;
    const { text } = this.props;

    return (
      <DivStyling>
        <Label text={text} />
        <SelectStyling>
          <option disabled value="select-value">
            -- Select --
          </option>
          {options.map(opt => (
            <option key={opt.text} value={opt.value}>
              {opt.text}
            </option>
          ))}
        </SelectStyling>
      </DivStyling>
    );
  }
}

Select.propTypes = {
  text: PropTypes.string,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.any,
      text: PropTypes.string,
    })
  ),
};

export default Select;
