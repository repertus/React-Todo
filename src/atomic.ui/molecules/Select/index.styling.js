import styled from 'styled-components';

export const SelectStyling = styled.select`
  display: block;
  width: 100%;
  font-family: 'Architects Daughter';
  border: 1px solid #ccc;
  border-radius: 4px;
  height: 24px;
  font-size: 14px;
  line-height: 1.42857143;
  color: #555;
  background-color: #fff;
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
  transition: border-color ease-in-out 0.15s, box-shadow ease-in-out 0.15s;
`;

export const DivStyling = styled.div`
  margin-top: 20px;
`;
