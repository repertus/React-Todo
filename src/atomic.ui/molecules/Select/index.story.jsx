import React from 'react';
import { storiesOf } from '@storybook/react';

import Select from './index';

const text = 'Label Text';
const options = [
  {
    value: 'a_high',
    text: 'A high priority item',
  },
  {
    value: 'b_medium',
    text: 'A medium priority item',
  },
  {
    value: 'c_low',
    text: 'A low priority item',
  },
];

storiesOf('Molecules', module).add('Select dropdown', () => (
  <Select text={text} options={options} />
));
