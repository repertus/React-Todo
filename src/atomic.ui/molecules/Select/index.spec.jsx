// Dependencies
import React from 'react';
import renderer from 'react-test-renderer';

// Component to test...
import Select from './index';

const text = 'Label Text';
const options = [
  {
    value: 'a_high',
    text: 'A high priority item',
  },
  {
    value: 'b_medium',
    text: 'A medium priority item',
  },
  {
    value: 'c_low',
    text: 'A low priority item',
  },
];

describe('Select molecule tests', () => {
  test('Render simple select', () => {
    const component = renderer.create(<Select text={text} options={options} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
