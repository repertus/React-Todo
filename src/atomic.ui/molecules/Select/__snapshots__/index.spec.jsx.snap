// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select molecule tests Render simple select 1`] = `
<div
  className="sc-bwzfXH bszZAY"
>
  <label
    className="sc-htpNat brBYsh"
  >
    Label Text
  </label>
  <select
    className="sc-bdVaJa jHHhRH"
  >
    <option
      disabled={true}
      value="select-value"
    >
      -- Select --
    </option>
    <option
      value="a_high"
    >
      A high priority item
    </option>
    <option
      value="b_medium"
    >
      A medium priority item
    </option>
    <option
      value="c_low"
    >
      A low priority item
    </option>
  </select>
</div>
`;
